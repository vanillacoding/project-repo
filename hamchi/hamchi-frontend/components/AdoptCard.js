import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import PhotoCard from './PhotoCard';
import colors from '../theme/color';

const AdoptCard = ({ data, onPress }) => {
  return (
    <View
      style={styles.cardContainer}>
      <PhotoCard
        uri={data.image}
        type={data.type}
      />
      <Text style={styles.text}>{data.name}</Text>
      <Text style={styles.username}>{data.ownerName}ë‹˜</Text>
      <View style={styles.divider} />
    </View>
  );
};

const styles = StyleSheet.create({
  cardContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  text: {
    alignSelf: 'flex-start',
    marginTop: 10,
    padding: 20,
    height: 20,
  },
  username: {
    alignSelf: 'flex-start',
    padding: 20,
    paddingTop: 6,
    height: 20,
  },
  divider: {
    alignSelf: 'center',
    width: '80%',
    height: 1,
    backgroundColor: colors.outline
  }
});

export default AdoptCard;
