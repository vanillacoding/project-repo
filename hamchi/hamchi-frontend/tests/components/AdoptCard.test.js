import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import AdoptCard from '../../components/AdoptCard';

describe('AdoptCard component test', () => {
  it('renders correctly', () => {
    const data = {
      name: 'name',
      ownerName: 'owner name'
    };
    const tree = renderer.create(<AdoptCard data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows text and username of data prop', async () => {
    const data = {
      name: 'name',
      ownerName: 'owner name'
    };

    const { getByText } = render(<AdoptCard data={data} />);
    const isNameExists = getByText(data.name);
    const isOwnerNameExists = getByText(`${data.ownerName}ë‹˜`);

    expect(isNameExists).toBeTruthy();
    expect(isOwnerNameExists).toBeTruthy();
  });
});
