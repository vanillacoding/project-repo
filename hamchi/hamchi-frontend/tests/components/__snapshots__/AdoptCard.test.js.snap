// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdoptCard component test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "top": 12,
          "width": "40%",
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#CCCCCC",
          "borderRadius": 10,
          "height": 150,
          "width": "90%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "aspectRatio": 1,
            "borderRadius": 10,
            "height": undefined,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "height": 20,
        "marginTop": 10,
        "padding": 20,
      }
    }
  >
    name
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "height": 20,
        "padding": 20,
        "paddingTop": 6,
      }
    }
  >
    owner name
    ë‹˜
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#CCCCCC",
        "height": 1,
        "width": "80%",
      }
    }
  />
</View>
`;
